packages/transaction-messages/src/transaction-message-size.ts

```diff
@@ -0,0 +1,16 @@
+import { BaseTransactionMessage } from './transaction-message';
+
+/**
+ * A type guard that checks if a transaction message is within the size limit
+ * when compiled into a transaction.
+ */
+export type TransactionMessageWithinSizeLimit = { readonly __within_size_limit: unique symbol };
```

Comment:
Consider how you're going to implement this. I'll throw out some thoughts but ultimately leave it up to you.

```ts
NominalType<'withinSizeLimit', true>;  // Either present or not

NominalType<'relationshipToSizeLimit', 'under' | 'meets' | 'over'>;  // Useful? I don't know.

NominalType<'size', 'within_1232'>;  // Imagine a world where the size limit gets increased. Do we need to be able to distinguish between limits at different versions of the cluster?
```

Reply:
I'm thinking:

```ts
NominalType<'transactionSize', 'withinLimit'>
```

As it follows a similar convention to the existing:

```ts
NominalType<'transactionSignedness', 'fullySigned'>
```

I don't think there is any point in using values such as `within_1232` since we are planning on fixing cluster difference at build time.